scripts/sysman/device.yml

```diff
@@ -355,6 +355,66 @@ returns:
     - $X_RESULT_ERROR_INVALID_SIZE:
         - "The provided value of pCount is not big enough to store information about all the processes currently attached to the device."
 
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Device Description"
+version: "1.8"
+class: $sDevice
+name: $s_device_description_t
+base: $s_base_desc_t
+members:
+    - type: $x_bool_t
+      name: "onSubdevice"
+      desc: "[out] True if the UUID belongs to the sub-device; false means that UUID belongs to the root device"
+    - type: uint32_t
+      name: "subdeviceId"
+      desc: "[out] If onSubdevice is true, this gives the ID of the sub-device"
+
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Retrieves sysman device and subdevice index for the given UUID and sysman driver"
+version: "1.8"
+class: $sDevice
+name: GetByUUID
+details:
+    - "API returns corresponding sysman root device handle and sub device Index for the given UUID and the driver handle." 
+    - "The application may call this function from simultaneous threads."
+    - "The implementation of this function should be lock-free."
+params:
+    - type: $s_driver_handle_t
+      name: "hDriver"
+      desc: "[in] handle of the sysman driver instance"
+    - type: $s_uuid_t
+      name: "uuid"
+      desc: "[in] universal unique identifier."           
+    - type: $s_device_handle_t*
+      name: "phDevice"
+      desc: "[out] handle of sysman device."
+    - type: $s_device_description_t*
+      name: "pDeviceDesc"
+      desc: "[out] pointer to the device description struct."
+            
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Retrieves UUID for the given sysman device handle and subdevice index"
+version: "1.8"
+class: $sDevice
+name: GetSubDeviceUUID
+details:
+    - "API returns the UUID for the given sysman device handle and subdevice index." 
```

Comment:
for a device with no subdevices, if user calls this API with subdeviceId = 0, then we return ERROR code or we return uuid of the root device?
it's more of a implementation detail, but just wanted to leave note here if we want to document behavior in prog guide somewhere

Reply:
with API name being "zesGetSubDeviceUUID", we can return proper error code if UUID does not belong to sub-device. what you suggest?.

Reply:
A subdevice and a device have the same type of handle. Maybe better is to have this more generic, as zeDeviceGetSubDeviceProperties, where we return an array of properties, and the size of the array would be the same as the number of sub-devices. That would allow us to have this extendable for the future.

Reply:
@jandres742, what you are suggesting is something like below.         
    zesDeviceGetSubDeviceProperties(zes_device_handle_t, uint32_t *pCount, zes_subdevice_properties_t*) where zes_subdevice_properties_t{stype,pnext, subDeviceID, uuid_t}. User would first find pCount value with first call,  allocate memory for  zes_subdevice_properties_t array and query sub device properties for all sub devices of the sysman handle. do you agree?

Reply:
@AshwinKumarKulkarni above proposal for `zesDeviceGetSubDeviceProperties` looks good, please modify as suggested

Reply:
Modified as per proposal.